### DISCUSSION TITLE
Developing formulae or casks

### DISCUSSION THREAD

**Discussion by snakebyte69** [Homebrew/discussions#5059] [Writing Formulae/Casks]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.4-26-g3ed42ac
ORIGIN: https://github.com/Homebrew/brew
HEAD: 3ed42ac320d8c3bcb5499aee2f1770b70c76ec28
Last commit: 7 hours ago
Core tap HEAD: 3ef7d3e13549bdb977d180600b77d2ae02962378
Core tap last commit: 6 hours ago
Core tap JSON: 16 Jan 08:11 UTC
Core cask tap JSON: 16 Jan 08:11 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.lSIPYBXGz5/org.xquartz:0
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.1.4 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: octa-core 64-bit icelake
Clang: 15.0.0 build 1500
Git: 2.39.3 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 14.2.1-x86_64
CLT: 15.1.0.0.1.1700200546
Xcode: 15.2
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libz80ex.1.1.21.dylib
  /usr/local/lib/libz80ex_dasm.1.1.21.dylib

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected header files:
  /usr/local/include/cpmio.h
  /usr/local/include/cpmredir.h
  /usr/local/include/z80ex/z80ex.h
  /usr/local/include/z80ex/z80ex_common.h
  /usr/local/include/z80ex/z80ex_dasm.h

Warning: Unbrewed static libraries were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected static libraries:
  /usr/local/lib/libcpmio.a
  /usr/local/lib/libcpmredir.a
  /usr/local/lib/libz80ex.a
  /usr/local/lib/libz80ex_dasm.a

Warning: You have uncommitted modifications to Homebrew/homebrew-core.
If this is a surprise to you, then you should stash these modifications.
Stashing returns Homebrew to a pristine state but can be undone
should you later need to do so for some reason.
  cd /usr/local/Homebrew/Library/Taps/homebrew/homebrew-core && git stash -u && git clean -d -f

Uncommitted files:
   M Formula/z/z80asm.rb
  ?? Formula/z/z80ex.rb

Warning: Some taps are not on the default git origin branch and may not receive
updates. If this is a surprise to you, check out the default branch with:
  git -C $(brew --repo snakebyte69/snake) checkout 
snake@Snakes-MacBook-Air ~ %
```

### Description of issue

class Z80ex < Formula
  homepage "https://github.com/snakebyte69/z80ex"
  url "https://github.com/snakebyte69/z80ex/releases/tag/ez80ex-brew/z80ex.tar.gz"
  version "1.1.21"
  sha256 "0066fa53b13e09b9c286b75c6d28de07412902b06cdef964e6c04f26e055d0a6"
  license "GPL-2.0-or-later"

  def install
    system "./configure", "--disable-silent-rules", *std_configure_args
    system "sudo make"
    system "sudo make install"
  end

  test do
    system "false"
  end
end

-------
% brew install snakebyte69/homebrew-snake/z80ex

Warning: No remote 'origin' in /usr/local/Homebrew/Library/Taps/snakebyte69/homebrew-snake, skipping update!
==> Fetching snakebyte69/snake/z80ex
==> Downloading https://github.com/snakebyte69/z80ex/releases/tag/ez80ex-brew/z8
Already downloaded: /Users/snake/Library/Caches/Homebrew/downloads/f988d5f3153f56bbddbdabcdd867410b18b496eaee5f6fa54a228d2487807cbe--z80ex.tar.gz
==> Installing z80ex from snakebyte69/snake
tar: Error opening archive: Unrecognized archive format
Error: Failure while executing; `/usr/bin/env tar --extract --no-same-owner --file /Users/snake/Library/Caches/Homebrew/downloads/f988d5f3153f56bbddbdabcdd867410b18b496eaee5f6fa54a228d2487807cbe--z80ex.tar.gz --directory /private/tmp/d20240116-63762-ujy9pf` exited with 1. Here's the output:
tar: Error opening archive: Unrecognized archive format

% brew install snakebyte69/homebrew-snake/z80ex
==> Fetching snakebyte69/snake/z80ex
==> Downloading https://github.com/snakebyte69/z80ex/releases/tag/ez80ex-brew/z80ex.tar.gz
Already downloaded: /Users/snake/Library/Caches/Homebrew/downloads/f988d5f3153f56bbddbdabcdd867410b18b496eaee5f6fa54a228d2487807cbe--z80ex.tar.gz
==> Installing z80ex from snakebyte69/snake
tar: Error opening archive: Unrecognized archive format
Error: Failure while executing; `/usr/bin/env tar --extract --no-same-owner --file /Users/snake/Library/Caches/Homebrew/downloads/f988d5f3153f56bbddbdabcdd867410b18b496eaee5f6fa54a228d2487807cbe--z80ex.tar.gz --directory /private/tmp/d20240116-64229-4ywewm` exited with 1. Here's the output:
tar: Error opening archive: Unrecognized archive format

% 

-----
Not sure why it doesn't like the archive
I get the same error with .tar.gz or .tar

**Comment by ZhongRuoyu** [comment]
https://github.com/snakebyte69/z80ex/releases/tag/ez80ex-brew/z80ex.tar.gz is an HTML page. You probably want https://github.com/snakebyte69/z80ex/archive/refs/tags/ez80ex-brew.tar.gz.
